src/transformers/models/whisper/modeling_whisper.py

```diff
@@ -2969,7 +2976,8 @@ def forward(
             )
 
         if self.config.use_weighted_layer_sum:
-            hidden_states = torch.stack(encoder_outputs, dim=1)
+            hidden_states = encoder_outputs[_HIDDEN_STATES_START_POSITION]
```

Comment:
Again, following the Wav2Vec2 design:
https://github.com/huggingface/transformers/blob/2c1eebc1216549d8195d7d1c6adb8b99afee3ec5/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L2104